assembly/bindgen.ts

```diff
@@ -224,9 +246,9 @@ function decode<T, V = Uint8Array>(buf: V, name: string = ""): T {
   }
   //@ts-ignore
   if (value instanceof u128) {
-    assert(val instanceof JSON.Str);
+    assert(val instanceof JSON.Str, "Value with Key: " + name + " expected type string to decode u128 but got " + JSONTypeToString(val));
```

Comment:
Any reason class name cannot be specified? Also I think displaying not just type of supplied value but the value itself can be helpful.

I also wonder if `assert` going to run string concatenation anyway or only if condition fails?

P.S. Wonder how much effort is it to implement https://github.com/AssemblyScript/assemblyscript/issues/466

Reply:
Added test that shows that the concatenation doesn't happen if `assert(true...`.  @ailisp and myself had worked on it, but there have been changes to the compiler and we would need to try again and make a real PR.

Reply:
Also the class name is u128, which is mentioned.  As for the value, it could be too much since it could be anything.